marklogic-data-hub-central/src/main/java/com/marklogic/hub/central/controllers/EntitySearchController.java

```diff
@@ -54,7 +57,22 @@ public String search(@RequestBody SearchQuery searchQuery, @RequestParam(default
     @RequestMapping(method = RequestMethod.GET)
     @ResponseBody
     public JsonNode getRecord(@RequestParam String docUri, @RequestParam(defaultValue = "final") String database) {
-        return getEntitySearchService(database).getRecord(docUri);
+        return getEntitySearchService(database).getRecord(docUri, true);
+    }
+
+    @RequestMapping(value = "/downloadRecord", method = RequestMethod.GET)
+    public void downloadRecord(@RequestParam String docUri, @RequestParam(defaultValue = "final") String database, HttpServletResponse response) {
+        String[] docUriArray = docUri.split("/");
+        String fileName = docUriArray[docUriArray.length-1];
+        response.setContentType(MediaType.APPLICATION_OCTET_STREAM_VALUE);
+        response.addHeader("Content-Disposition", String.format("attachment; filename=%s", fileName));
+
+        try (OutputStream out = response.getOutputStream()) {
+            newEntitySearchManager(database).downloadRecord(docUri, out);
+            response.flushBuffer();
+        } catch (IOException e) {
+            throw new RuntimeException(String.format("Unable to download record with docUri: %s with cause: %s", docUri, e.getMessage()));
```

Comment:
Use "URI" in the error message ; "docUri" is a variable name that is not as meaningful to a user like Pari as "URI" is.